operator-framework-core/src/main/java/io/javaoperatorsdk/operator/processing/event/ReconciliationDispatcher.java

```diff
@@ -173,7 +173,9 @@ public int getAttemptCount() {
 
           @Override
           public boolean isLastAttempt() {
-            return controller.getConfiguration().getRetry() == null;
+            // on first try, we can only rely on the configured behavior
+            // if enabled, will at least produce one RetryExecution
+            return !controller.getConfiguration().getRetry().isPresent();
```

Comment:
This needs to be optimized to avoid multiple call to configuration doing initialization each time